ballerina/googlesheets_client.bal

```diff
@@ -0,0 +1,520 @@
+                            rowArray[columnName] = <time:TimeOfDay>typedValue;
+                        } else if typedValue is time:Civil {
+                            rowArray[columnName] = <time:Civil>typedValue;
+                        } else if typedValue is time:Utc {
+                            rowArray[columnName] = <time:Utc>typedValue;
+                        }
+                    } else {
+                        (string|int|decimal|time:Date|time:TimeOfDay|time:Civil|time:Utc)|error typedValue = self.dataConverter(value, dataType);
+                        if typedValue is error {
+                            return <Error>error(typedValue.message());
+                        }
+                        rowArray[columnName] = <(string|int|decimal)>typedValue;
+                    }
+                    i = i + 1;
+                }
+                rowTable.push(rowArray);
+
+            }
+            return rowTable.toStream();
+        } else {
+            return <Error>error(textResponse.message());
+        }
+    }
+
+    # Performs an SQL `UPDATE` operation to update multiple entity records in the database.
+    #
+    # + key - the key of the entity
+    # + updateRecord - the record to be updated
+    # + return - `()` if the operation is performed successfully.
+    # A `ForeignKeyConstraintViolationError` if the operation violates a foreign key constraint.
+    # A `persist:Error` if the operation fails due to another reason.
+    public isolated function runUpdateQuery(anydata key, record {} updateRecord) returns Error? {
+        string[] entityKeys = self.fieldMetadata.keys();
+        (int|string|decimal|boolean|float)[] values = [];
+        if (key is string|int|decimal|float) {
+            sheets:DeveloperMetadataLookupFilter filter = {locationType: "ROW", metadataKey: self.tableName, metadataValue: key.toString()};
+            sheets:ValueRange[]|error rows = self.googleSheetClient->getRowByDataFilter(self.spreadsheetId, self.sheetId, filter);
+            if rows is error {
+                return <Error> error(rows.message());
+            }
+            if (rows.length() == 0) {
+                return <Error>error(string `No element found for given key: ${key.toString()}`);
+            } else if rows.length() > 1 {
+                return <Error>error(string `Multiple elements found for given key: ${key.toString()}`);
+            }
+            foreach string entityKey in entityKeys {
+                if !updateRecord.hasKey(entityKey) && (self.keyFields.indexOf(entityKey) != ()) {
+                    values.push(key);
+                } else if !updateRecord.hasKey(entityKey) && (self.keyFields.indexOf(entityKey) == ()) {
+                    int? indexOfKey = self.fieldMetadata.keys().indexOf(entityKey, 0);
+                    if (indexOfKey !is ()) {
+                        values.push(rows[0].values[indexOfKey]);
+                    }
```

Comment:
```suggestion
                    int indexOfKey = <int>self.fieldMetadata.keys().indexOf(entityKey, 0);
                    values.push(rows[0].values[indexOfKey]);
```

Reply:
updated in [4346c39](https://github.com/ballerina-platform/module-ballerina-persist/pull/201/commits/4346c3959a465aeb05de0872b94ea1056a715a9a)